Eviction/chunk lru

2a46m4 opened the pull request.

ijnslc committed:
implement chunk eviction when all zones are full

ijnslc committed:
Merge branch 'main' of github.com:johnramsden/ZNWorkload into eviction/chunk-lru

ijnslc committed:
add documentation

ijnslc committed:
use NULL

johnramsden committed:
Merge branch 'main' into eviction/chunk-lru

johnramsden commented:
Right now I'm hitting an assertion:

https://github.com/johnramsden/ZNCache/blob/04a939cfa307d4159e1fa4d860746194ecaca3e5/src/cachemap.c#L140

johnramsden commented on src/cachemap.c:
```diff
@@ -101,13 +102,62 @@ zn_cachemap_find(struct zn_cachemap *map, const uint32_t data_id) {
+	assert(res);
+	assert(res->type == RESULT_LOC);
+	assert(res->value.location.chunk_offset == chunk_offset);
+	assert(res->value.location.id == data_id);
```
Always zero for some reason

johnramsden committed:
Fix missing id, add zone_map print

johnramsden committed:
Merge main into eviction/cache-lru

Commit:
use posix_memalign

Commit:
enforce only one thread evicting at a time

This is done by returning immediately when the lock fails. Also fix a small bug with Chunk
